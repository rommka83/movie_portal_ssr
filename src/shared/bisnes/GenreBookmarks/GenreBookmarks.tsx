import React, { FC } from 'react';
import styles from './genrebookmarks.module.css';
import { nanoid } from '@reduxjs/toolkit';
import { type } from 'os';
import { HTMLAttributes } from 'react';
import classNames from 'classnames';
import Link from 'next/link';

interface IProps {
  home?: boolean;
  page?: string;
  ganre: { name: string }[];
}
export const GenreBookmarks: FC<HTMLAttributes<HTMLUListElement> & IProps> = ({
  ganre,
  className,
  home,
  page,
}) => {
  return (
    <ul className={classNames(styles.list, className)}>
      {home && (
        <li className={styles.item} key={nanoid()}>
          <Link href='/' className={styles.link}>
            мой иви
          </Link>
        </li>
      )}
      {ganre.map((el) => {
        return (
          <li className={styles.item} key={nanoid()}>
            <a href='https://www.ivi.ru/movies' className={styles.link}>
              {el.name}
            </a>
          </li>
        );
      })}
      {page && (
        <li className={styles.item} key={nanoid()}>
          <span className={classNames(styles.link, styles.lastLink)}>{page}</span>
        </li>
      )}
    </ul>
  );
};
