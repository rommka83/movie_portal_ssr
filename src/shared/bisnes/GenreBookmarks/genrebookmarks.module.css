.list {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}
.item {
  display: flex;
  align-items: center;
}
.item:not(:first-child)::before {
  content: '';
  display: inline-block;
  width: 2px;
  height: 2px;
  background-color: var(--gray-72);
  margin-left: 7px;
  margin-right: 7px;
}
.link {
  color: var(--white);
  font-size: 15px;
  line-height: 20px;
  font-weight: 500;
  opacity: 0.5;
  transition: opacity 0.3s ease-in-out;
}
.link:hover {
  opacity: 1;
}
.lastLink {
  opacity: 1;
}
