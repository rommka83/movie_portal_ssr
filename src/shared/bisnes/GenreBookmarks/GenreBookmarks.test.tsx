import renderer from 'react-test-renderer';
import { GenreBookmarks } from './GenreBookmarks';

describe('проверка GenreBookmarks', () => {
  test('отрисовка GenreBookmarks ', () => {
    const component = renderer.create(<GenreBookmarks ganre={[{ name: 'string' }]} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
