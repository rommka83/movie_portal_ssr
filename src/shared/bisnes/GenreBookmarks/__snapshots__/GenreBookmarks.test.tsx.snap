// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка GenreBookmarks отрисовка GenreBookmarks  1`] = `
<ul
  className="list"
>
  <li
    className="item"
  >
    <a
      className="link"
      href="https://www.ivi.ru/movies"
    >
      string
    </a>
  </li>
</ul>
`;
